src/goto-programs/assign_params_as_non_det.cpp

```diff
@@ -0,0 +1,59 @@
+#include <goto-programs/assign_params_as_non_det.h>
+
+bool assign_params_as_non_det::runOnFunction(
+  std::pair<const dstring, goto_functiont> &F)
+{
+  if (context.find_symbol(F.first) == nullptr)
+    return false; // Not exist
+
+  exprt symbol = symbol_expr(*context.find_symbol(F.first));
+
+  if (!symbol.type().is_code())
+    return false; // Not expected
+  code_typet t = to_code_type(symbol.type());
+
+  if (symbol.name().as_string() != target_function)
+    return false; // Not target function
+
+  if (!F.second.body_available)
+    return false; // Empty function
+
+  /*
+    Foreach parameter, create an assignment to nondet value and insert in the front
+    E.g. func(int x, bool y) {...}
+    =>
+    func(int x, bool y)  { x = nondet_int(); y = nondet_bool(); ...}
+  */
+  goto_programt &goto_program = F.second.body;
+  auto it = (goto_program).instructions.begin();
+
+  for (const auto &arg : t.arguments())
+  {
+    // lhs
+    const auto &_id = arg.get("#identifier");
+    if (context.find_symbol(_id) == nullptr)
+      return false; // Not expected
+    exprt lhs = symbol_expr(*context.find_symbol(_id));
```

Comment:
can you check if the symbol returned by `context.find_symbol(_id)` has the location of the parameter declaration?

Reply:
Thanks for the suggestion. The location is now
```sh
[Counterexample]


State 1 file 1.c line 3 column 1 thread 0
----------------------------------------------------
  x = 0 (00000000 00000000 00000000 00000000)

State 2 file 1.c line 5 column 2 function foo thread 0
----------------------------------------------------
  x = 1 (00000000 00000000 00000000 00000001)

State 3 file 1.c line 6 column 2 function foo thread 0
----------------------------------------------------
Violated property:
  file 1.c line 6 column 2 function foo
  assertion x > 3
  x > 3
```